src/main/java/org/nypl/findawayclientlibrary/DownloadService.java

```diff
@@ -141,19 +183,253 @@ public void onNext(Integer progress) {
+    return eventsSubscriptionStatus;
   }
 
 
 
   /* ------------------------------------ DOWNLOAD EVENT HANDLERS ------------------------------------- */
+
+  /**
+   * Translate DownloadEngine-specific status flags into ones our app uses.
+   *
+   * @param contentId
+   * @return
+   */
+  public Integer getDownloadStatus(String contentId) {
+    // GIGO
+    if (contentId == null) {
+      return DOWNLOAD_ERROR;
+    }
+
+    try {
+      // TODO:  used to throw ContentNotFoundException.  test that feeding junk contentId doesn't break the new DownloadEngine.
+      if (downloadEngine.getStatus(contentId).equals(DownloadStatus.NOT_DOWNLOADED)) {
+        return DOWNLOAD_NEEDED;
+      }
+
+      if (downloadEngine.getStatus(contentId).equals(DownloadStatus.QUEUED) || downloadEngine.getStatus(contentId).equals(DownloadStatus.DOWNLOADED)) {
+        return DOWNLOAD_RUNNING;
+      }
+
+      if (downloadEngine.getStatus(contentId).equals(DownloadStatus.PAUSED)) {
+        return DOWNLOAD_PAUSED;
+      }
+
+      if (downloadEngine.getStatus(contentId).equals(DownloadStatus.DOWNLOADED)) {
+        return DOWNLOAD_SUCCESS;
+      }
+    } catch (Exception e) {
+      // was our contentId not recognized?  record the error, and return.
+      LogHelper.e(TAG, e, "DownloadEngine cannot give status for contentId=", contentId);
+    }
+
+    // unrecognized status, something's wrong
+    return DOWNLOAD_ERROR;
+  }
+
+
+  /**
+   * Make a download request, and ask the Findaway download engine to fulfill it.
+   * Will download the entire book, starting with the chapter specified, skipping any
+   * chapters that are already downloaded.  Will wrap to also download the beginning of the book,
+   * if needed.  So, if pass in chapter 5, then will download chapters 5 - end, and then
+   * download chapters 1-4.
+   *
+   * NOTE: Audio files are downloaded and stored under the application's standard internal files directory. This directory is deleted when the application is removed.
+   *
+   * NOTE:  As long as we use the DownloadEngine to download, queue, delete, etc., the Engine will have stored knowledge
+   * of the current state in the findaway internal database.  So, if I use the engine to delete, it will change the status to “NOT_DOWNLOADED”
+   * which I can check for and re-download later.
+   *
+   * NOTE:  Call getDownloadEngine().getProgress(contentId) to get the download progress
+   * of a book as a percentage from 0 to 100.  Or use the download event system.
+   *
+   * TODO:  Test how to request downloads for 2different books (may need to make my own queue).
+   *
+   * @param contentId  the catalog book id
+   * @param contentId  id of the license to play book (what was checked out to patron)
+   * @param chapter  the chapter to start download at
+   * @param part  the more specific chapter part to start download at
+   */
+  public Integer downloadAudio(String contentId, String licenseId, Integer chapter, Integer part) {
```

Comment:
We discussed putting these fields into a model instead of passing the fields everywhere

Reply:
Yep, I spun off a branch for it.